dbms/src/Functions/FunctionsLogical.h

```diff
@@ -216,7 +214,8 @@ struct AssociativeOperationImpl
             //   then, the vec will be:
             //      vec = {1, 0, 2} (error, we only want 0 or 1)
             // See issue: https://github.com/pingcap/tidb/issues/37258
-            return Op::isSaturatedValue(a) ? !!a : continuation.apply(i);
+            bool a = static_cast<bool>(vec[i]);
+            return Op::isSaturatedValue(a) ? static_cast<UInt8>(a) : continuation.apply(i);
```

Comment:
I think just return `a` is ok since compiler will handle it properly.

Reply:
> I think just return `a` is ok since compiler will handle it properly.

okk